test/ViewExtensionLibraryTests/DynamoLibraryItemsTests.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Text;
+using CefSharp;
+using Dynamo;
+using Dynamo.LibraryUI.Handlers;
+using Moq;
+using Newtonsoft.Json;
+using NUnit.Framework;
+
+namespace ViewExtensionLibraryTests
+{
+    public class DynamoLibraryItemsTests : DynamoModelTestBase
+    {
+        protected override string GetUserUserDataRootFolder()
+        {
+            return Path.GetTempPath();
+        }
+
+        protected override void GetLibrariesToPreload(List<string> libraries)
+        {
+            libraries.Add("DSCoreNodes.dll");
+            base.GetLibrariesToPreload(libraries);
+        }
+
+        [Test]
+        [Category("UnitTests")]
+        public void VerifyIconsForLibraryItems()
+        {
+            var iconProvider = new IconResourceProvider(GetModel().PathManager);
+            var request = new Mock<IRequest>();
+            var resource = "ViewExtensionLibraryTests.resources.libraryItems.json";
+            using (var s = Assembly.GetExecutingAssembly().GetManifestResourceStream(resource))
+            {
+                var data = GetLoadedTypesFromJson(s);
+                var types = data.loadedTypes;
+
+                var unresolved = new List<LoadedTypeItem>();
+                foreach (var item in types)
+                {
+                    request.Setup(r => r.Url).Returns(item.iconUrl);
+                    string extension = "txt";
```

Comment:
Minor. The definition of extension can be moved out of the for loop and initialized with an empty string.